src/Nethermind/Nethermind.Core/Block.cs

```diff
@@ -113,6 +113,8 @@ public Transaction[] Transactions
 
     [JsonIgnore]
     public ArrayPoolList<AddressAsKey>? AccountChanges { get; set; }
+    [JsonIgnore]
+    internal volatile int TransactionProcessed;
```

Comment:
Can we pass it in different way? Block should be more of domain object.
Similar problem is with AccountChanges.

Reply:
Is a bit messy because of where the compents surface where currently the only shared comms is the block.

Will think about if it can be improved though

Reply:
lets introduce some sharable block processing context